src/components/Keyword/KeywordList/index.js

```diff
@@ -0,0 +1,329 @@
+    const [keywordSearch,setKeywordSearch] = useState('');
+    const [goStore, setGoStore] = useState(false);
+    const [deleteList, setDeleteList] = useState(false);
+
+    const inputSearch = useRef(null);
+
+    const [isToggle,setIsToggel] = useState(false);
+
+    const onClickMenu = useCallback((e)=>{        
+        const menu = e.target.innerText;
+        setMenu(menu);
+        setCheckAll(false);
+        setReadNotification(false);
+        setNotReadNotification(false);
+        setCheckListId([]);
+
+        if(list.length===0){
+            return;
+        }
+
+        if(menu === '전체'){
+            setList(keywordList);
+        }else if(menu === '아우누리'){
+            const filterList = keywordList.filter((item)=>{
+                if(item.url.includes(AUNURI)){
+                    return item;
+                }
+            })
+
+            setList(filterList);
+        }else if(menu === '아우미르'){
+            const filterList = keywordList.filter((item)=>{
+                if(item.url.includes(AOUMIR)){
+                    return item;
+                }
+            })
+
+            setList(filterList);
+        }
+
+    },[keywordList]);
+
+    const onClickAllSelect = useCallback(()=>{
+
+        setCheckAll((prev)=>!prev);
+        
+    },[list,checkAll]);
+
+    const onClickReadNotification = useCallback(()=>{
+        setReadNotification((prev)=>!prev);
+        setNotReadNotification(false);
+    },[notReadNotification]);
+
+    const onClickNotReadNotification = useCallback(()=>{
+        setNotReadNotification((prev)=>!prev);
+        setReadNotification(false);
+    },[readNotification]);
+
+    const onChangeKeywordSearch = useCallback((e)=>{
+        setKeywordSearch(e.target.value);
+    },[keywordSearch]);
+
+    const onClickSearch = useCallback((e)=>{    
+        
+        const filterList = list.filter((item)=>{
+            
+            if(`${item.title}`.includes(`${keywordSearch}`)){
+                return item;
+            }
+        });
+
+        setList(filterList);
+        setKeywordSearch('');
+        inputSearch.current.focus();
+    },[list,keywordSearch]);
+
+    const onClickGoStore = useCallback(()=>{
+        setGoStore(true);
+    },[]);
+
+    const onClickDeleteList = useCallback(()=>{
+        setDeleteList(true);
+    },[]);
+
+    const onClickCheckSome = useCallback((id)=>{
+
+        let newCheckListId = [...checkListId];
+
+        if(checkListId.includes(id)){
+            
+            newCheckListId = checkListId.filter((item)=>{
+                if(item !== id){
+                    return item;
+                }
+            })
+
+            setCheckListId(newCheckListId);
+        }else{
+            newCheckListId.push(id);
+            setCheckListId(newCheckListId);
+        }
+
+    },[checkListId]);
+
+    const onClickReadItem = useCallback((id,isRead)=>{
+        if(!isRead){
+            dispatch(readKeywordItem(id));
+        }
```

Comment:
`early return case ` 가 더 깔끔해보입니다

```suggestion
        if(isRead) return;
        
        dispatch(readKeywordItem(id));
```